app/src/main/java/com/endava/flights/MainFlightViewModel.kt

```diff
@@ -0,0 +1,22 @@
+package com.endava.flights
+
+import androidx.lifecycle.ViewModel
+import com.endava.flights.datasource.FlightImpl
+import com.endava.flights.model.GetFlightsByServicesUseCase
+import com.endava.flights.model.Service
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class MainFlightViewModel : ViewModel() {
+    private lateinit var getCheapestFlightsUseCase : GetFlightsByServicesUseCase
+
+    suspend fun getCheapestFlight(searchingWords : List<Service>) : List<String> {
+
+        getCheapestFlightsUseCase = GetFlightsByServicesUseCase(
+            searchingWords, FlightImpl()
+        )
+        return withContext(Dispatchers.IO){
```

Comment:
Be careful about where you want to change the coroutine dispatcher. Should be here? Take into account you are doing it on the Repository Impl

Reply:
But maybe it's ok if you just do the mock repository, See comment below